activesupport/test/fork_tracker_test.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+require "abstract_unit"
+
+class ForkTrackerTest < ActiveSupport::TestCase
+  def test_object_fork
+    read, write = IO.pipe
+    handler = ActiveSupport::ForkTracker.after_fork { write.write "forked" }
```

Comment:
I'd maybe throw another local in here:

```suggestion
    called = false
    handler = ActiveSupport::ForkTracker.after_fork do
      called = true
      write.write "forked"
    end
```

.. and then assert that's still false, to prove it was executed in the child only, and not the parent?